bundle/src/main/java/com/adobe/acs/commons/dam/impl/AssetsFolderPropertiesSupport.java

```diff
@@ -0,0 +1,247 @@
+        ),
+        @Property(
+                name = "sling.filter.scope",
+                value = "REQUEST",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.filter.pattern",
+                value = "/content/dam/.*",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.methods",
+                value = "GET",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.resourceTypes",
+                value = "acs-commons/touchui-widgets/asset-folder-properties-support",
+                propertyPrivate = true
+        )
+})
+@Service
+public class AssetsFolderPropertiesSupport extends SlingSafeMethodsServlet implements Filter, SlingPostProcessor {
+    private static final Logger log = LoggerFactory.getLogger(AssetsFolderPropertiesSupport.class);
+
+    private static final String DAM_PATH_PREFIX = "/content/dam";
+    private static final String POST_METHOD = "post";
+    private static final String OPERATION = ":operation";
+    private static final String DAM_FOLDER_SHARE_OPERATION = "dam.share.folder";
+    private static final String GRANITE_UI_FORM_VALUES = "granite.ui.form.values";
+
+    /**
+     * This method is responsible for post processing POSTs to the FolderShareHandler PostOperation (:operation = dam.share.folder).
+     * This method will store a whitelisted set of request parameters to their relative location off of the [sling:*Folder] node.
+     *
+     * Note, this is executed AFTER the OOTB FolderShareHandler PostOperation.
+     *
+     * At this time this method only supports single-value Strings and ignores all @typeHints.
+     *
+     * This method must fail fast via the accepts(...) method.
+     *
+     * @param servletRequest the request object
+     * @param servletResponse the response object
+     * @param chain the filter chain
+     * @throws IOException
+     * @throws ServletException
+     */
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
+        final SlingHttpServletRequest request = (SlingHttpServletRequest) servletRequest;
+        final SlingHttpServletResponse response = (SlingHttpServletResponse) servletResponse;
+
+        if (!accepts(request)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        log.trace("ACS AEM Commons Assets Folder Properties Support applied to POST Request");
+        chain.doFilter(new AssetsFolderPropertiesSupportRequest(request, null), response);
+    }
+
+    public void init(FilterConfig filterConfig) throws ServletException {
+        // Do Nothing
+    }
+
+    public void destroy() {
+        // Do Nothing
+    }
+
+    @Reference(target="&(sling.post.operation=dam.share.folder)(sling.servlet.methods=POST)")
+    private PostOperation folderShareHandler;
```

Comment:
I usually declare all my references at the top.  Is that not a standard pattern?